packages/terra-form-select/src/Select.jsx

```diff
@@ -0,0 +1,285 @@
+  name: null,
+  required: false,
+  defaultValue: undefined,
+  value: undefined,
+  disabled: false,
+  isInvalid: false,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class Select extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.getInitialState = this.getInitialState.bind(this);
+    this.setTargetRef = this.setTargetRef.bind(this);
+    this.getTargetRef = this.getTargetRef.bind(this);
+    this.handleOnClick = this.handleOnClick.bind(this);
+    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+    this.handleOnRequestClose = this.handleOnRequestClose.bind(this);
+    this.handleSelection = this.handleSelection.bind(this);
+    this.wrapOnClick = this.wrapOnClick.bind(this);
+    this.wrapOnKeyDown = this.wrapOnKeyDown.bind(this);
+    this.state = {
+      selectedValue: this.getInitialState(),
+      isOpen: false,
+    };
+  }
+
+  getInitialState() {
+    if (!this.props.value) {
+      const childArray = React.Children.toArray(this.props.children);
+      for (let i = 0; i < childArray.length; i += 1) {
+        const child = childArray[i];
+        if (child.props.isSelected || child.props.value === this.props.defaultValue) {
+          return child.props.value;
+        }
+      }
+    }
+    return '';
+  }
+
+  setTargetRef(node) {
+    this.targetRef = node;
+  }
+
+  getTargetRef() {
+    return this.targetRef;
+  }
+
+  wrapOnClick(option) {
+    return (event) => {
+      this.handleSelection(event, option);
+
+      if (option.props.onClick) {
+        option.props.onClick(event);
+      }
+    };
+  }
+
+  wrapOnKeyDown(option) {
+    return (event) => {
+      if (option.props.onKeyDown) {
+        option.props.onKeyDown(event);
+      }
+
+      if (event.nativeEvent.keyCode === SelectUtils.KEYCODES.SPACE || event.nativeEvent.keyCode === SelectUtils.KEYCODES.ENTER) {
+        this.handleSelection(event, option);
+      } else if (event.nativeEvent.keyCode === SelectUtils.KEYCODES.UP_ARROW) {
+        let previousFocus = event.target.previousSibling;
+        while (previousFocus.tabIndex < 0) {
+          previousFocus = previousFocus.previousSibling;
+          if (!previousFocus) {
+            return;
+          }
+        }
+        previousFocus.focus();
+      } else if (event.nativeEvent.keyCode === SelectUtils.KEYCODES.DOWN_ARROW) {
+        let nextFocus = event.target.nextSibling;
+        while (nextFocus.tabIndex < 0) {
+          nextFocus = nextFocus.nextSibling;
+          if (!nextFocus) {
+            return;
+          }
+        }
+        nextFocus.focus();
+      }
+    };
+  }
+
+  handleSelection(event, option) {
+    if (this.props.onChange && option.props.value !== this.state.value) {
+      this.props.onChange(event, option.props.value);
```

Comment:
There was a short discussion a while back about flipping the order of these props: https://github.com/cerner/terra-core/pull/768#discussion_r143198019

With this being a new package it might be worth discussing if we want to make this change or continue exposing the event as the first param

Reply:
We talked about this in scrum this morning and there was a consensus that we should continue with our current pattern of passing the event as the first argument.